diff --git a/node_modules/@slack/socket-mode/dist/SocketModeClient.js b/node_modules/@slack/socket-mode/dist/SocketModeClient.js
index 2967682..48560a5 100644
--- a/node_modules/@slack/socket-mode/dist/SocketModeClient.js
+++ b/node_modules/@slack/socket-mode/dist/SocketModeClient.js
@@ -316,7 +316,7 @@ class SocketModeClient extends eventemitter3_1.EventEmitter {
                 const flatMessage = JSON.stringify(message);
                 this.logger.debug(`Sending a WebSocket message: ${flatMessage}`);
                 this.websocket.send(flatMessage, (error) => {
-                    if (error !== undefined) {
+                    if (error !== undefined && error != null) {
                         this.logger.error(`Failed to send a WebSocket message (error: ${error.message})`);
                         return reject((0, errors_1.websocketErrorWithOriginal)(error));
                     }
